### DISCUSSION TITLE
2023/05/22 10:10:36 database disk image is malformed (11)

### DISCUSSION THREAD

**Discussion by MuhammedKilinc60** [pocketbase/pocketbase#2530] [Q&A]
Hello guys I need help. 

I try to serve but get this error 

2023/05/22 10:10:36 database disk image is malformed (11)

**Comment by ganigeorgiev** [comment]
Please provide as much details as possible for your setup so that I can try to reproduce it.

- What PocketBase version do you use (you can run `./pocketbase --version`)?

- How PocketBase is deployed and on what OS?

- Do you have Docker? If yes, make sure that the mounted volume is properly configured.

- Do you have other application that operates on the `pb_data/data.db` file?

- Did you manually move/copy the database while the application was running? If yes, did you also copy the auxiliary `-wal` file?

**Reply by MuhammedKilinc60** [reply]
Hello thanks for your Response 
What PocketBase version do you use?:0.13

How PocketBase is deployed and on what OS? Ubuntu

Do you have Docker? If yes, make sure that the mounted volume is properly configured. No 

Do you have other application that operates on the pb_data/data.db file? no

Did you manually move/copy the database while the application was running? If yes, did you also copy the auxiliary -wal file? I made a git merge and I think it built mess 


I solved the problem with a backup(thanks to god) but I have a question about pocketbase now
since the data is saved in pb_data it also gets copied in commits etc. but I always have some problems 

1. since it is a running app i cannot terminate it and everytime i add and commit the pb files fastly change and i cannot commit since there are new uncommited files. 
2. and even if i terminate and push, i cannot merge since there are so many files like data or logs which collide with each other. 
3. even if i add pb_data into .gitignore there is the other problem that i dont commit it and if i do git fetch i lose my data

**Reply by ganigeorgiev** [reply]
> I made a git merge and I think it built mess

Yes, it is possible. The `pb_data` directory usually shouldn't be part of your git repo. I recommend to add it in your `.gitignore` OR at least commit the `pb_data` only when the application is not running to avoid corruptions (especially if multiple people works on the same project).

If you want your collections configuration (_without the records data_) as part of your git repo, you can safely commit the `pb_migrations` directory.

**Reply by MuhammedKilinc60** [reply]
oh so if i commit pb_migrations only it saves the data aswell ?

**Reply by ganigeorgiev** [reply]
`pb_migrations` contains all the necessary information to build your collections (schema fields, indexes, rules, etc.) from scratch but it doesn't contain your app data like locally uploaded files, created records (_unless the record is created in a migration_), etc.

**Reply by MuhammedKilinc60** [reply]
yeah but I would like to save my data in the git repo aswel, so if i dont understand it wrong, it is not really possible

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand your use case but **it is not safe to use `git pull/merge` with `pb_data`** because multiple users/environments can make changes to the same db file (note that the db changes are not consisting only of appending the data to the file, some of them could change the file at the beginning, in the middle, etc. and when multiple such changes collide it could result in db corruption).

It is safe to commit your `pb_data` _only_ if you are the only one that makes changes to the database (**and always before applying new change you have to make sure to run `git pull`**).

**Reply by aluemku** [reply]
Try to use sqlite backup command ... sqlite3 <path to "pb_data/data.db"> '.backup <backups/data.backup>', put <pb_data> to .gitignore but leave backup directory in the git repository

**Reply by ganigeorgiev** [reply]
@aluemku Note that the db backup file still could get corrupted if multiple users commit the same db backup because `git merge` doesn't just replace the old file with the new one (_unless `--ours` or `--theirs` option is used_).

Also note that with [v0.16.0](https://github.com/pocketbase/pocketbase/releases/tag/v0.16.0) if you just want a backup of your pb_data you can generate it from the Admin UI > Settings > Backups.
